### DISCUSSION TITLE
loadfile?

### DISCUSSION THREAD

**Discussion by akavel** [mlua-rs/rlua#226] [Q&A]
In the standard C Lua API, luaL_loadfile (or alternatively lua_load + lua_Reader) allows loading a file without having to first read it whole into memory, as a "stream" of "blocks" (calls to lua_Reader). Is it possible to load a Lua file in luar in a streamed way, without first reading it whole into memory? I didn't seem to be able to find such a feature, which made me somewhat surprised; I'm curious if there are some reasons for this, or it just was not really added yet? It's not that I desperately need it at the moment, but I admit I'd like to be able to make a mental note as to what's the maintainers position on this? Or if I just missed something obvious?

**Comment by jugglerchris** [comment]
Hi,
No, there isn't currently a way of doing directly in the rlua API, although you could call the Lua `load` function if you had to.

I think a variant of `Context::load()` which took a `T: Read` sounds like it would be a good addition to me.
Chris
